rancher/rancher
# Bump Rancher Machine to v0.15.0-rancher106

## Issue: https://github.com/rancher/rancher/issues/42972

 
## Problem

v0.15.0-rancher106 of Rancher Machine fixes a bug where rotating the vSphere username in cloud credentials does not take effect for existing clusters as Rancher Machine continues using the old username. More details here: https://github.com/rancher/machine/pull/228#discussion_r1344763787.
 
## Solution

Rancher Machine version bump.
 
## Testing


## Engineering Testing
### Manual Testing

Covered by unit tests (and I don't have access to the new vSphere yet).

### Automated Testing

* Test types added/modified:
    * Unit: Wrote unit tests to check that username rotation works here: https://github.com/rancher/machine/pull/228/files#diff-8c671a925ab240bbb797aea3f416a4ca00af2f901a4f81b8075a584cb0730e10R12
  

Summary: Wrote unit tests to check that username rotation works here: https://github.com/rancher/machine/pull/228/files#diff-8c671a925ab240bbb797aea3f416a4ca00af2f901a4f81b8075a584cb0730e10R12

## QA Testing Considerations

Just need to make sure that the following works:
1. Create a vSphere cluster
2. Change the username in the associated cloud credential to a different valid username
3. Try delete the cluster. It should delete just fine 

### Regressions Considerations

None

Existing / newly added automated tests that provide evidence there are no regressions:
* Unit and integration tests in rancher/machine
